nym-vpn-core/crates/nym-vpn-account-controller/src/storage/account.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2024 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+use std::sync::Arc;
+
+use nym_vpn_api_client::types::{Device, VpnApiAccount};
+use nym_vpn_store::{mnemonic::Mnemonic, VpnStorage};
+
+use crate::error::Error;
+
+#[derive(Debug, Clone)]
+pub(crate) struct AccountStorage<S>
+where
+    S: VpnStorage,
```

Comment:
any reason for using generic rather than concrete type? unless, of course, there are some tests ; )

Reply:
IIRC it's because we'll likely need custom storage for mobile later on